### PR TITLE
Translate Zoom control tooltips and autocomplete results

### PR DESCRIPTION
**Pull request by fbonzon**
See the **[live demo](http://xn--franois-xxa.ch/graphhopper/?locale=fr-CH&layer=OpenStreetMap)**

The menu with autocomplete suggestions now displays geographic names in the same language as the other UI elements. As mentioned in the [documentation](https://graphhopper.com/api/1/docs/geocoding/#url-parameters), currently only French (fr), English (en), German (de) and Italian (it) are supported. Other languages will continue to display English names.

Translation strings for "zoom in" and "zoom out" have already been committed earlier today.

### CODE REVIEW

File: web/src/main/webapp/js/autocomplete.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var formatTools = require('./tools/format.js');
 var GHInput = require('./graphhopper/GHInput.js');
+var GHRoute = require('./graphhopper/GHRoute.js');
```
Comment (fbonzon) [inline web/src/main/webapp/js/autocomplete.js]: Fixing unrelated bug here. This is required for line 71 below.

File: web/src/main/webapp/js/autocomplete.js [hunk 2]
```diff
@@ -81,7 +83,9 @@ AutoComplete.prototype.createGeocodeURL = function (ghRequest, prevIndex) {
     var path = this.createPath(this.host + "/geocode?limit=6&type=" + this.dataType + "&key=" + this.key);
     if (prevIndex >= 0 && prevIndex < ghRequest.route.size()) {
         var point = ghRequest.route.getIndex(prevIndex);
-        path += "&point=" + point.lat + "," + point.lng;
+        if (point.isResolved()) {
```
Comment (fbonzon) [inline web/src/main/webapp/js/autocomplete.js]: avoids `point=undefined,undefined` in the URL

**Comment by karussell** [comment]
Thanks!
Finally this old bug gets resolved :) #272

File: web/src/main/webapp/js/translate.js [hunk 3]
```diff
@@ -6,11 +6,11 @@ var mathTools = require('./tools/math.js');
 
 function tr2(key, args) {
     if (key === null) {
-        log("ERROR: key was null?");
+        console.log("ERROR: key was null?");
         return "";
     }
     if (defaultTranslationMap === null) {
-        log("ERROR: defaultTranslationMap was not initialized?");
```
Comment (karussell) [inline web/src/main/webapp/js/translate.js]: Thanks, it looks like `log(` is no longer defined (?)

We should then fix the several other places too (maybe another issue)

File: web/src/main/webapp/js/main-template.js [hunk 4]
```diff
@@ -158,6 +159,8 @@ $(document).ready(function (e) {
 
                 // execute query
                 initFromParams(urlParams, true);
+
+                checkInput();
```
Comment (karussell) [inline web/src/main/webapp/js/main-template.js]: Is this necessary for the `undefined` bug?

File: web/src/main/webapp/js/translate.js (hunk 3 above)
Comment (fbonzon) [inline web/src/main/webapp/js/translate.js]: Correct, log() is undefined. There are more instances, but are all commented out, so I did not fix those.

File: web/src/main/webapp/js/main-template.js (hunk 4 above)
Comment (fbonzon) [inline web/src/main/webapp/js/main-template.js]: No, unrelated. This has to do with the order of things happening.

When the ready() function is executed, the autocomplete menu for the first 2 locations fields (empty From and To) is already finished, and still has "en" locale. One must wait until the API call to retrieve translations to know the locale.

**Comment by fbonzon** [comment]
#272 is not solved here. It is a different problem unfortunately.

I am solving bad URLs sent to geocoding API, that had a bad location bias. Inspect network traffic in the browser developper console. It happens for example with the 2nd field ("To") when both location fields are initially empty.

File: web/src/main/webapp/js/main-template.js (hunk 4 above)
Comment (fbonzon) [inline web/src/main/webapp/js/main-template.js]: The checkInput(); on line 207 below now happens before the new instance here. It becomes redundant, and could probably be removed.

**Pull request merged**